app/services/solidus_square/payment_sync_service.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+module SolidusSquare
+  class PaymentSyncService < Base
+    attr_reader :params
+
+    def initialize(params)
+      @params = params
+
+      super()
+    end
+
+    def call
+      create_square_payment! if hosted_checkout?
+
+      update_payment_source!
+      payment.complete if payment_source_captured?
```

Comment:
`payment.complete` return false if the transaction makes some error. 
It should be replaced with `payment.complete!` so it will raise an exception

Reply:
https://github.com/nebulab/solidus_square/pull/43/commits/abea86ce4b352a0e46b124beeecf97797b17300c